rfc/0000-keyless-job-auth.rst

```diff
@@ -0,0 +1,204 @@
+===============================================
+Keyless Cloud Authentication in Hail Batch Jobs
+===============================================
+
+.. author:: Daniel Goldstein
+.. date-accepted:: Leave blank. This will be filled in when the proposal is accepted.
+.. ticket-url:: Leave blank. This will eventually be filled with the
+                ticket URL which will track the progress of the
+                implementation of the feature.
+.. implemented:: Leave blank. This will be filled in with the first Hail version which
+                 implements the described feature.
+.. header:: This proposal is `discussed at this pull request <https://github.com/hail-is/hail-rfc/pull/0>`_.
+            **After creating the pull request, edit this file again, update the
+            number in the link, and delete this bold sentence.**
+.. sectnum::
+.. contents::
+.. role::
+
+In a Hail Batch pipeline, all user code runs under one of two identities associated
+with the underlying cloud platform. On the user's computer, code runs as the user's
+human identity. Inside a Batch job running in the cloud, the job is running as a
+Hail-managed identity created for that user on signup. We will refer to this
+identities as the user's Human Identity and the Robot Identity, respectively.
+
+In the current system, Batch jobs are given access to the users' robot identity
+through a file planted in the job's filesystem containing the identity's
+credentials. In GCP, the credentials are a GSA key file. In Azure, the credentials
+are a username/password for a AAD Service Principal. These credentials are created,
```

Comment:
FWIW, Azure Active Directory has been renamed (to, afaict, everyone's great anger and frustration) Microsoft Entra ID.

Reply:
Begrudgingly changed